packages/react-dom/src/__tests__/ReactMount-test.js

```diff
@@ -160,6 +169,303 @@ describe('ReactMount', () => {
     );
   });
 
+  it('should warn when a hydrated element has inner text mismatch', () => {
+    // See fixtures/ssr: ssr-warnForTextDifference
+
+    class Component extends React.Component {
+      render() {
+        return this.props.children;
+      }
+    }
+
+    const div = document.createElement('div');
+    const markup = ReactDOMServer.renderToString(
+      <Component>
+        <div>server text</div>
+      </Component>,
+    );
+    div.innerHTML = markup;
+
+    expect(() =>
+      ReactDOM.hydrate(
+        <Component>
+          <div>client text</div>
+        </Component>,
+        div,
+      ),
+    ).toWarnDev(
+      'Text content did not match. ' +
+        'Server: "server text" ' +
+        'Client: "client text"\n' +
+        '    in div (at **)\n' +
+        '    in Component (at **)',
+    );
+  });
+
+  it('should warn when hydrating a text node over a mismatching text node', () => {
+    // See fixtures/ssr: ssr-warnForTextDifference-didNotMatchHydratedContainerTextInstance
+
+    const div = document.createElement('div');
+    const markup = ReactDOMServer.renderToString('server text');
+    div.innerHTML = markup;
+
+    expect(() => ReactDOM.hydrate('client text', div)).toWarnDev(
+      'Text content did not match. ' +
+        'Server: "server text" ' +
+        'Client: "client text"',
+      // Without the component stack here because it's empty: rendering a text node directly into the root node.
+      {withoutStack: true},
```

Comment:
> Without the component stack here because it's empty: rendering a text node directly into the root node

A thought came to me here: why doesn't React show in the stack the place where the `render` or `hydrate` was called? This would be 1) consistent, there'd always be a stack; 2) useful for apps with multiple render entry points.